src/test/modules/compiler/compiler.js

```diff
@@ -0,0 +1,78 @@
+/*globals describe, it*/
+let assert = require('assert');
+
+// TODO: need to rethink i18n and how that is required in each module
+require('../../../lib/core/i18n/i18n');
+
+let Compiler = require('../../../lib/modules/compiler');
+let File = require('../../../lib/core/file.js');
+let Plugins = require('../../../lib/core/plugins.js');
+let TestLogger = require('../../../lib/utils/test_logger');
+let Events = require('../../../lib/core/events');
+
+let readFile = function(file) {
+  return new File({filename: file, type: File.types.dapp_file, path: file});
+};
+
+const currentSolcVersion = require('../../../../package.json').dependencies.solc;
+const TestEvents = {
+  request: (cmd, cb) => {
+    cb(currentSolcVersion);
+  },
+  emit: (_ev, _data) => {}
+};
```

Comment:
For if you have to do other tests, you can use `sinon` instead. It let's you stub the methods like you just did, but also gives you a spy, so you can check how many times a function was called and with which arguments.